### ISSUE TITLE
Confusing error message when a Literal float without decimal digits is assigned to int

### ISSUE THREAD

**Issue by viblo** [microsoft/pyright#1078]
**Describe the bug**
If a literal is set as a float with no decimal digits, such as `1.0` and its used where a int is expected the error message is quite confusing, since the `.0` is not included in the error. In many times its easy to find the problem, but sometimes the literal is defined far away from its usage, and then it becomes more difficult to spot the error.

**Expected behavior**
It would be great if the error could be a little bit more descriptive, either with some hint what the type of the Literal is, or that the Literal itself includes `.0`. 

**Screenshots or Code**

```python
x = 15.0
y: int = x
```
Error message: 
```
Expression of type "Literal[15]" cannot be assigned to declared type "int"
  "Literal[15]" is incompatible with "int"
```
**VS Code extension or command-line**
Running pyright from VS Code with Pylance 2020.9.8.

**Comment by erictraut** [comment]
Thanks for the bug report. Pyright shouldn't be treating any float or complex value as a Literal type. PEP 586 clearly indicates that neither of these are valid Literal types. So `x` in your example should have an inferred type of `float`, not `Literal[15.0]`. This will be fixed in the next release.

**Comment by viblo** [comment]
Thanks for the quick response and fix!

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.77, which I just published. It will also be included in the next release of Pylance.
